agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -397,6 +452,7 @@ func (state *DockerTaskEngineState) RemoveTask(task *apitask.Task) {
 		// remove v3 endpoint mappings
 		state.removeV3EndpointIDToTaskContainerUnsafe(dockerContainer.Container.V3EndpointID)
 	}
+	delete(state.taskToPulledContainer, task.Arn)
```

Comment:
we should add unit test case validating the remove.

Reply:
`TestRemoveTask` in  dockerstate_test.go is updated to validate this behavior. Thanks!